pkg/apis/batch/validation/validation.go

```diff
@@ -168,6 +170,58 @@ func validateJobSpec(spec *batch.JobSpec, fldPath *field.Path, opts apivalidatio
 		}
 	}
 
+	if spec.PodFailurePolicy != nil {
+		containerNames := sets.NewString()
+		for _, containerSpec := range spec.Template.Spec.Containers {
+			containerNames.Insert(containerSpec.Name)
+		}
+		for _, containerSpec := range spec.Template.Spec.InitContainers {
+			containerNames.Insert(containerSpec.Name)
+		}
+		for i, rule := range spec.PodFailurePolicy.Rules {
+			rulePath := fldPath.Child("podFailurePolicy").Index(i)
+
+			if rule.Action != batch.PodFailurePolicyActionIgnore && rule.Action != batch.PodFailurePolicyActionFailJob && rule.Action != batch.PodFailurePolicyActionCount {
+				actionPath := rulePath.Child("action")
+				allErrs = append(allErrs, field.NotSupported(actionPath, rule.Action, []string{string(batch.PodFailurePolicyActionIgnore), string(batch.PodFailurePolicyActionFailJob), string(batch.PodFailurePolicyActionCount)}))
+			}
+			if rule.OnExitCodes != nil {
+				onExitCodesPath := rulePath.Child("onExitCodes")
+				if rule.OnExitCodes.Operator != batch.PodFailurePolicyOnExitCodesOpIn && rule.OnExitCodes.Operator != batch.PodFailurePolicyOnExitCodesOpNotIn {
+					path := onExitCodesPath.Child("operator")
+					allErrs = append(allErrs, field.NotSupported(path, rule.OnExitCodes.Operator, []string{string(batch.PodFailurePolicyOnExitCodesOpIn), string(batch.PodFailurePolicyOnExitCodesOpNotIn)}))
+				}
+				if rule.OnExitCodes.ContainerName != nil && !containerNames.Has(*rule.OnExitCodes.ContainerName) {
+					path := onExitCodesPath.Child("containerName")
+					allErrs = append(allErrs, field.Invalid(path, *rule.OnExitCodes.ContainerName, "must be one of the container names in the pod template"))
+				}
+				if rule.OnExitCodes.Operator == batch.PodFailurePolicyOnExitCodesOpIn {
+					for j, exitCodeValue := range rule.OnExitCodes.Values {
+						if exitCodeValue == 0 {
+							path := onExitCodesPath.Child("values").Index(j)
+							allErrs = append(allErrs, field.Invalid(path, exitCodeValue, "must not be 0 for the In operator"))
+						}
+					}
+				}
+			}
+			if rule.OnPodConditions != nil {
+				onPodConditionsPath := rulePath.Child("onPodConditions")
+				for j, pattern := range rule.OnPodConditions {
+					patternPath := onPodConditionsPath.Index(j)
+					allErrs = append(allErrs, apivalidation.ValidateQualifiedName(string(pattern.Type), patternPath.Child("type"))...)
+				}
+			}
+			if rule.OnExitCodes != nil && rule.OnPodConditions != nil {
+				ruleJsonBytes, err := json.Marshal(rule)
+				if err != nil {
+					klog.ErrorS(err, "Failed to marshal pod failure policy rule", "rule", rule)
+				} else {
+					allErrs = append(allErrs, field.Invalid(rulePath, string(ruleJsonBytes), "Specifying both OnExitCodes and OnPodConditions is not supported"))
+				}
+			}
+		}
```

Comment:
I don't see validation that at least one of `onExitCodes` and `onPodConditions` was specified

Reply:
Added